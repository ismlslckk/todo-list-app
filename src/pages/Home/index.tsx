const Home = () => (

  <>
    Home
    <br />
  </>
);

export default Home;
