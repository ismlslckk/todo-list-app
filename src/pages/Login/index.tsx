import { InnerWrapper, Wrapper } from '@/components/atoms';

const Form = () =>
/*   const [form, setForm] = useState({
    username: '',
    password: '',
  });

  const handleChange = (e:any) => {
    const value = e.target.type === 'checkbox' ? e.target.checked : e.target.value;
    setForm((state) => ({
      ...state,
      [e.target.name]: value,
    }));
  };

  const handleSubmit = (e:any) => {
    e.preventDefault();
  }; */

  // eslint-disable-next-line implicit-arrow-linebreak
  (
    <Wrapper>
      <InnerWrapper>
        login page . .
      </InnerWrapper>
    </Wrapper>
  );
export default Form;
