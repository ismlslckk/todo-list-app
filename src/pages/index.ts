export {default as Home} from '@pages/Home';
export {default as NotFound} from '@pages/NotFound';
export {default as Login} from '@pages/Login';
export {default as Register} from '@pages/Register';
