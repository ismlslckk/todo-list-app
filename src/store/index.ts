import { configureStore } from '@reduxjs/toolkit';
import { TypedUseSelectorHook, useDispatch, useSelector } from 'react-redux';
import todoSlice from '@/features/todo/todoSlice';
import globalSlice from '@/features/globalSlice';

const store = configureStore({
  reducer: {
    todos: todoSlice,
    globalState: globalSlice,
  },
});

export default store;

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;

export const useAppDispatch = () => useDispatch<AppDispatch>();
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;
