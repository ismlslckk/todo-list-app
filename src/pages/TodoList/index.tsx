import { useEffect, useState } from 'react';
import { TodoItem } from '@/components';
import { InnerWrapper, Wrapper } from '@/components/atoms';
import { useAppDispatch, useAppSelector } from '@/store';
import {
  Todo, TodoType,
} from '@/types';
import styles from './todolist.module.scss';
import { clearCompleted, setSelectedTodoType } from '@/features/todo/todoSlice';

const TodoList = () => {
  const todoState = useAppSelector((state) => state.todoState);
  const { todos, selectedTodoType } = todoState;

  const dispatch = useAppDispatch();

  const [leftItemCount, setLeftItemCount] = useState(0);

  const [listedRecords, setListedRecords] = useState<Todo[]>([]);

  const calculateLeftItemCount = () => {
    setLeftItemCount(listedRecords.filter((todo:Todo) => !todo.completed).length);
  };

  const listAllRecors = () => {
    dispatch(setSelectedTodoType(TodoType.ALL));
    setListedRecords([...todos]);
  };

  const listActiveRecors = () => {
    dispatch(setSelectedTodoType(TodoType.ACTIVE));
    const records = [...todos.filter((item:Todo) => item.completed === false)];
    setListedRecords(records);
  };

  const listCompletedRecors = () => {
    dispatch(setSelectedTodoType(TodoType.COMPLETED));
    const records = [...todos.filter((item:Todo) => item.completed === true)];
    setListedRecords(records);
  };

  const clearCompletedHandle = () => {
    dispatch(clearCompleted());
    dispatch(setSelectedTodoType(TodoType.ACTIVE));
  };

  const actions:any = {
    [`${TodoType.ALL}`]: listAllRecors,
    [`${TodoType.ACTIVE}`]: listActiveRecors,
    [`${TodoType.COMPLETED}`]: listCompletedRecors,
  };

  useEffect(() => {
    dispatch(setSelectedTodoType(TodoType.ALL));
  }, []);

  useEffect(() => {
    calculateLeftItemCount();
  }, [listedRecords]);

  useEffect(() => {
    if (selectedTodoType) {
      actions[selectedTodoType]();
    }
  }, [selectedTodoType, todos]);

  return (
    <Wrapper>
      <InnerWrapper>
        <div className={styles.todoListBody}>
          <div>
            {listedRecords.map((item) => (<TodoItem key={item.id} todo={item} />))}
          </div>
        </div>
        <div className={styles.footer}>
          <span className={styles.leftItemCount}>{`${leftItemCount} items left`}</span>
          <button onClick={() => listAllRecors()} type="button" className={styles.clickable}>All</button>
          <button onClick={() => listActiveRecors()} type="button" className={styles.clickable}>Active</button>
          <button onClick={() => listCompletedRecors()} type="button" className={styles.clickable}>Completed</button>
          <button onClick={() => clearCompletedHandle()} type="button" className={styles.clickable}>Clear Completed</button>
        </div>
      </InnerWrapper>
    </Wrapper>
  );
};

export default TodoList;
