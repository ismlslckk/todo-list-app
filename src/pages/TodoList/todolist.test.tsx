import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '@/store';
import { TodoList } from '@/pages';

test('expect todo list page is rendering with todo types', async () => {
  render(<Provider store={store}><TodoList /></Provider>);

  expect(screen.queryByText('All')).not.toBeNull();
  expect(screen.queryByText('Active')).not.toBeNull();
  expect(screen.queryByText('Completed')).not.toBeNull();
  expect(screen.queryByText('Clear Completed')).not.toBeNull();
});
