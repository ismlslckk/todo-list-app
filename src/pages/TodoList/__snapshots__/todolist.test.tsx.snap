// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checkbox component tests renders correctly 1`] = `
<div>
  <div
    className="todoListBody"
    data-rbd-droppable-context-id="1"
    data-rbd-droppable-id="draggable-todo-list"
  />
  <div
    className="footer"
  >
    <span
      className="leftItemCount"
    >
      0 items left
    </span>
    <button
      className="clickable selectedItem"
      onClick={[Function]}
      type="button"
    >
      All
    </button>
    <button
      className="clickable"
      onClick={[Function]}
      type="Button"
    >
      Active
    </button>
    <button
      className="clickable"
      onClick={[Function]}
      type="Button"
    >
      Completed
    </button>
    <button
      className="clickable"
      onClick={[Function]}
      type="button"
    >
      Clear Completed
    </button>
  </div>
</div>
`;
